### DISCUSSION TITLE
Can Patchwork be used to replace the exit method?

### DISCUSSION THREAD

**Discussion by BlakeCThompson** [antecedent/patchwork#132] [General]
I want to overwrite the exit method so that I can test my controller response. Is this something I can do with patchwork?

**Comment by antecedent** [comment]
Yes, it is possible.

In the same directory that your entry script is in, create a `patchwork.json` with the following contents:

```json
{
    "redefinable-internals": [
        "die",
        "exit"
    ]
}
```

Once you have it, proceed with the redefining as usual:

```php
foreach (['die', 'exit'] as $construct) {
    Patchwork\redefine($construct, function() {
        // ...
    });
}
```

Please let me know if you need some more assistance.
